exchanges/okx/okx.go

```diff
@@ -0,0 +1,4312 @@
+	if err != nil {
+		return nil, err
+	}
+	if len(resp) == 1 {
+		return &resp[0], nil
+	}
+	return nil, errNoValidResponseFromServer
+}
+
+// GetOrderList retrieves all incomplete orders under the current account.
+func (ok *Okx) GetOrderList(ctx context.Context, arg *OrderListRequestParams) ([]OrderDetail, error) {
+	if arg == nil {
+		return nil, errNilArgument
+	}
+	params := url.Values{}
+	if arg.InstrumentType != "" {
+		params.Set("instType", strings.ToUpper(arg.InstrumentType))
+	}
+	if arg.InstrumentID != "" {
+		params.Set("instId", arg.InstrumentID)
+	}
+	if arg.Underlying != "" {
+		params.Set("uly", arg.Underlying)
+	}
+	if arg.OrderType != "" {
+		params.Set("orderType", strings.ToLower(arg.OrderType))
+	}
+	if arg.State != "" {
+		params.Set("state", arg.State)
+	}
+	if !arg.Before.IsZero() {
+		params.Set("before", strconv.FormatInt(arg.Before.UnixMilli(), 10))
+	}
+	if !arg.After.IsZero() {
+		params.Set("after", strconv.FormatInt(arg.After.UnixMilli(), 10))
+	}
+	if arg.Limit > 0 {
+		params.Set("limit", strconv.FormatInt(arg.Limit, 10))
+	}
+	var resp []OrderDetail
+	return resp, ok.SendHTTPRequest(ctx, exchange.RestSpot, getOrderListEPL, http.MethodGet, common.EncodeURLValues(pendingTradeOrders, params), nil, &resp, true)
+}
+
+// Get7DayOrderHistory retrieves the completed order data for the last 7 days, and the incomplete orders that have been cancelled are only reserved for 2 hours.
+func (ok *Okx) Get7DayOrderHistory(ctx context.Context, arg *OrderHistoryRequestParams) ([]OrderDetail, error) {
+	return ok.getOrderHistory(ctx, arg, tradeHistory, getOrderHistory7DaysEPL)
+}
+
+// Get3MonthOrderHistory retrieves the completed order data for the last 7 days, and the incomplete orders that have been cancelled are only reserved for 2 hours.
+func (ok *Okx) Get3MonthOrderHistory(ctx context.Context, arg *OrderHistoryRequestParams) ([]OrderDetail, error) {
+	return ok.getOrderHistory(ctx, arg, orderHistoryArchive, getOrderHistory3MonthsEPL)
+}
+
+// getOrderHistory retrieves the order history of the past limited times
+func (ok *Okx) getOrderHistory(ctx context.Context, arg *OrderHistoryRequestParams, route string, rateLimit request.EndpointLimit) ([]OrderDetail, error) {
+	params := url.Values{}
+	if arg.InstrumentType != "" {
+		params.Set("instType", strings.ToUpper(arg.InstrumentType))
+	} else {
+		return nil, errMissingRequiredArgInstType
+	}
+	if arg.InstrumentID != "" {
+		params.Set("instId", arg.InstrumentID)
+	}
+	if arg.Underlying != "" {
+		params.Set("uly", arg.Underlying)
+	}
+	if arg.OrderType != "" {
+		params.Set("orderType", strings.ToLower(arg.OrderType))
+	}
+	if arg.State != "" {
+		params.Set("state", arg.State)
+	}
+	if !arg.Before.IsZero() {
+		params.Set("before", strconv.FormatInt(arg.Before.UnixMilli(), 10))
+	}
+	if !arg.After.IsZero() {
+		params.Set("after", strconv.FormatInt(arg.After.UnixMilli(), 10))
+	}
```

Comment:
```suggestion
	if arg.Before != "" {
		params.Set("before", arg.Before)
	}
	if arg.After != "" {
		params.Set("after", arg.After)
	}
	if !arg.Start.IsZero() {
		params.Set("begin", strconv.FormatInt(arg.Start.UnixMilli(), 10))
	}
	if !arg.End.IsZero() {
		params.Set("end", strconv.FormatInt(arg.End.UnixMilli(), 10))
	}
```
To work with the recent observations